### ISSUE TITLE
Failed to create ACL rule when ACL table is not created

### ISSUE THREAD

**Issue by stcheng** [Azure/sonic-swss#209]

**Comment by andriymoroz-mlnx** [comment]
What if by mistake table will never be created?
Need to check whether consumer logic isn't broken if some doTask() will leave a record in consumer.m_toSync.
Are there any cases in other daemons we skip erasing some entries from consumer.m_toSync?

**Comment by stcheng** [comment]
yes. there any many examples that some tasks are not erased but waiting for dependencies. this fix is added because different tables are processed independently without guaranteeing the orders across different tables. it is possible that rules are processed before tables are processed and rules will be lost.

later callback logics will be added to replace the current retry logic.

**Comment by liatgrozovik** [comment]
I don't think that from the user perspective and rule should be created before it creates the table. table should be first created and only then rules should be added. this also goes on the delete flow. you should delete a table only if all the rules associated to it will be removed first.

**Comment by stcheng** [comment]
Let us first do the quick fix and then have a further discussion on what is the best way to resolve the above scenario.
